src/researches/passiveVoice/morphological/determinePassiveSentence.js

```diff
@@ -0,0 +1,49 @@
+const filter = require( "lodash/filter" );
+const getWords = require( "../../../stringProcessing/getWords.js" );
+
+// Verb-form lists per language
+const getPassiveVerbsRussian = require( "../../russian/passiveVoice/participlesShortenedList.js" )().all;
+
+/**
+ * Matches the sentence against passive verbs.
+ *
+ * @param {string} sentence The sentence to match against.
+ * @param {Array} passiveVerbs The array containing passive verb-forms.
+ * @returns {Array} The found passive verbs.
+ */
+let matchPassiveVerbs = function( sentence, passiveVerbs ) {
+	return filter( passiveVerbs, function( word ) {
+		let sentenceWords = getWords( sentence );
```

Comment:
Building it like this means that the words are extracted from the sentence every iteration of the `filter` loop. If you move the `let sentenceWords` out of the loop then that won't happen.

Reply:
Agreed, fixed.